pages/events.tsx

```diff
@@ -0,0 +1,231 @@
+    location: "online",
+    description:
+      "Come learn about how to get started with thirdweb's GamingKit and then put your skills to the test in a boundary-pushing web3 gaming hackathon! This event is perfect for beginners who want to learn about game development, or for experienced developers who want to try out our new tools. We'll have prizes for the most innovative and seamless games created during the hackathon, so come ready to showcase your talents!",
+    link: "https://lu.ma/rp3kickoff",
+  },
+  {
+    type: "Workshop",
+    title: "Getting Started with thirdweb | Ready Player 3 Hackathon",
+    timestamp: "2023-01-17T13:00:00-05:00",
+    location: "online",
+    description:
+      "Join us for a workshop on getting started with thirdweb, and then put your new skills to the test in our Ready Player 3 Hackathon!",
+    link: "https://lu.ma/rp3gettingstarted",
+  },
+  {
+    type: "Workshop",
+    title:
+      "Create AI Generated In-Game NFT Assets with Scenario.gg | Ready Player 3 Hackathon",
+    timestamp: "2023-01-18T13:00:00-05:00",
+    location: "online",
+    description:
+      "Come learn how to use the new AI game asset generation tool from scenario.gg in combination with thirdweb's GamingKit!",
+    link: "https://lu.ma/rp3scenario",
+  },
+  {
+    type: "Workshop",
+    title:
+      "thirdweb & Coinbase Cloud Code-Along: Build With GamingKit | Ready Player 3 Hackathon",
+    timestamp: "2023-01-19T13:00:00-05:00",
+    location: "online",
+    description:
+      "​Learn how to build a web3 game, powered by thirdweb's GamingKit!",
+    link: "https://lu.ma/rp3gamecodealong",
+  },
+  {
+    type: "Workshop",
+    title: "Workshop with Spindl.xyz | Ready Player 3 Hackathon",
+    timestamp: "2023-01-23T13:00:00-05:00",
+    location: "online",
+    description: "",
+    link: "https://lu.ma/rp3spindl",
+  },
+  {
+    type: "Workshop",
+    title: "Fireside Chat with Fractal | Ready Player 3 Hackathon",
+    timestamp: "2023-01-24T13:00:00-05:00",
+    location: "online",
+    description: "",
+    link: "https://lu.ma/rp3fractal",
+  },
+  {
+    type: "Workshop",
+    title:
+      "How to Create Your Hackathon Submission on DevPost + Q&A | Ready Player 3 Hackathon",
+    timestamp: "2023-01-25T13:00:00-05:00",
+    location: "online",
+    description: "",
+    link: "https://lu.ma/rp3submissions",
+  },
+  {
+    type: "Workshop",
+    title: "Ready Player 3 Hackathon | Closing Ceremony + Winners Announcement",
+    timestamp: "2023-02-06T13:00:00-05:00",
+    location: "online",
+    description: "",
+    link: "https://lu.ma/rp3closing",
+  },
+];
+
+const EventsPage: ThirdwebNextPage = () => {
+  const [sortBy, setSortBy] = useState<"date" | "location" | "type" | "title">(
+    "date",
+  );
+  const [sortedEvents, setSortedEvents] = useState(allEvents);
+
+  const handleSort = (value: "date" | "location" | "type" | "title") => {
+    setSortBy(value);
+    switch (value) {
+      case "date":
```

Comment:
date should probably be the default filter

filtering by title makes little sense to me

Reply:
instead of having filters we can always sort by title and have a "search" box in addition that filters by location, type and title based on string matching?

Reply:
so it would search for everything? 🤔

Reply:
Maybe we could do like a search using the title and add sort using location, type, etc.?

Reply:
what's the sorting parameter for "location"? how close the event is to the user? alphabetical makes no sense, that's not sort but grouping then.
same question for type

Reply:
"search" everything means the user can type "san francisco" and they see everything in san francisco, which makes sense, or they can type "hackathon" and they'll see all hackathons, or they can search a specific title and they'll find that, aka it gets users to success faster

Reply:
hmm yea that makes sense

Reply:
so search just using the title?